src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -75,12 +84,30 @@
   private static final String HADOOP_HOME_DOC = "The Hadoop home directory.";
   private static final String HADOOP_HOME_DISPLAY = "Hadoop home directory";
 
+  // Storage group
+  public static final String TOPIC_REGEX_CAPTURE_GROUP_CONFIG = "topic.regex.capture.group";
+  public static final String TOPIC_REGEX_CAPTURE_GROUP_DISPLAY = "Topic Regex Capture Group";
+  public static final String TOPIC_REGEX_CAPTURE_GROUP_DOC = "A regex that matches the entire "
+      + "topic and  specifies what groups to capture in the topic, so when specifying `${1}` in "
+      + "`topics.dir`, `${1}` will refer to the first captured group. Example config value of "
+      + "`([a-zA-Z]*)_([a-zA-Z]*)` match a topic that is two words delimited by a - and will "
+      + "capture both words as separate groupds, so for `topic.dir = ${1}/${2}` and "
+      + "`topic = example_name` the corresponding `topic.dir` will be `example/name/`. By default, "
+      + "this functionality is not enabled.";
+  public static final String TOPIC_REGEX_CAPTURE_GROUP_DEFAULT = null;
+
+  private static final String DIR_REGEX_DOC = " Supports ``${topic}`` in the value, which will be "
```

Comment:
`${0}` and `${topic}` would be the same if the regex matched the whole topic , which we could choose to enforce. I guess the cost of adding `${topic}` is pretty low, and this feature could be enabled without ever specifying a capture config like `^.*$`

Reply:
I didn't really wanna support `${0}`, `${topic}` is more intuitive I think